### PR TITLE
[ASCII-1984] send metric when there are changes between current PR and baseline

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

We currently send metrics for all the dependencies and external dependencies https://github.com/DataDog/datadog-agent/blob/a85c49ce5d919f511e7df258392cad3acffaf948/.gitlab/source_test/golang_deps_diff.yml#L53

Would be interesting to also sent the direct change of dependencies on each PR if there any. 

The PR enhances the diff task to send the metric so we can improve our dependency dashboard 😄 

### Motivation

Be able to track the dependencies changes on PR using a DD dashboard

### CODE REVIEW

File: tasks/diff.py [hunk 1]
```diff
@@ -138,14 +159,30 @@ def go_deps(ctx, baseline_ref=None, report_file=None):
...
                     goos, goarch = GOOS_MAPPING.get(platform), GOARCH_MAPPING.get(arch)
                     target = f"{binary}-{goos}-{goarch}"
                     prettytarget = f"{binary} {goos}/{goarch}"
 
+                    tags = [
+                        f"build:{build}",
+                        f"flavor:{flavor.name}",
+                        f"os:{goos}",
+                        f"arch:{goarch}",
+                        f"git_ref:#{commit_sha}",
```
Comment (pgimalac) [inline tasks/diff.py:174]: In the metric defined in `go_deps.py` we use `f"git_sha:{git_sha}",`, we might as well keep the same tag name (`git_sha`) and remove the `#` to be coherent ?

Comment (GustavoCaso) [reply tasks/diff.py:174]: Nice catch 👍

**Review by pgimalac — APPROVED** [review]
LGTM

**Review by pgimalac — COMMENTED** [review]
Note that since this job doesn't run on main we won't have metrics comparing one commit on main to the next...

File: tasks/diff.py [hunk 2]
```diff
@@ -138,14 +158,29 @@ def go_deps(ctx, baseline_ref=None, report_file=None):
...
                     target = f"{binary}-{goos}-{goarch}"
                     prettytarget = f"{binary} {goos}/{goarch}"
 
+                    tags = [
+                        f"build:{build}",
+                        f"flavor:{flavor.name}",
+                        f"os:{goos}",
+                        f"arch:{goarch}",
+                        f"git_sha:{commit_sha}",
+                    ]
```
Comment (FlorentClarret) [inline tasks/diff.py:174]: I think it would make more sense to have it in the `if report_metrics:` block

**Review by FlorentClarret — APPROVED** [review]
LGTM with one tiny suggestion

**Review by amenasria — APPROVED** [review]
**👏 praise:** ‏LGTM ! Nice one !

**Pull request merged**